entproto/internal/todo/ent/proto/entpb/entpb_user_service.go

```diff
@@ -342,3 +346,87 @@ func (svc *UserService) Delete(ctx context.Context, req *DeleteUserRequest) (*em
 	}
 
 }
+
+// List implements UserServiceServer.List
+func (svc *UserService) List(ctx context.Context, req *ListUserRequest) (*ListUserResponse, error) {
+	type token struct {
+		Id int
+	}
+	var (
+		err      error
+		entList  []*ent.User
+		pageSize int
+	)
+	pageSize = int(req.GetPageSize())
+	switch {
+	case pageSize < 0:
+		return nil, status.Errorf(codes.InvalidArgument, "page size cannot be less than zero")
+	case pageSize == 0 || pageSize > entproto.MaxPageSize:
+		pageSize = entproto.MaxPageSize
+	}
+	listQuery := svc.client.User.Query().
+		Order(ent.Desc(user.FieldID)).
+		Limit(pageSize + 1)
+	if req.GetPageToken() != nil {
+		buf := bytes.Buffer{}
+		bytes, err := base64.StdEncoding.DecodeString(req.PageToken.GetValue())
+		if err != nil {
```

Comment:
the code for encoding / decoding the token seems like it can be simplified. 
why do we need so many layers (a local token struct, a byte buffer, gob, etc...)
our goal is to have a base64 representation of the id of the last element we read, right? can this be done with less?